src/momento/internal/_utilities/_data_validation.py

```diff
@@ -43,6 +53,14 @@ def _list_as_bytes(
     return iter([_as_bytes(value) for value in values])
 
 
+def _set_as_bytes(
+    elements: TSetElementsInput, error_message: Optional[str] = DEFAULT_SET_CONVERSION_ERROR
+) -> TSetElementsInputBytes:
+    if not isinstance(elements, collections.abc.Iterable):
+        raise InvalidArgumentException(f"{error_message}{type(elements)}")
+    return iter({_as_bytes(element) for element in elements})
```

Comment:
Why return `iter` as opposed to building a set?

Reply:
If grpc takes an iterator it will (presumably) stream instead of copying. Could potentially save a lot of memory.

Reply:
Line 61 already builds a set object, so you're already paying that cost. You could change the function to be a generator if that's what you want:

```python
def set_elements_to_bytes(elements: List[bytes]) -> Iterator[bytes]:
    # check for collection type here
    for element in elements:
         yield _as_bytes(element)
```

Reply:
I do, but this iterator idea is going to take some more thought.